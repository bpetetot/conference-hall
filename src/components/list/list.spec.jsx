/* eslint-env jest */
import React from 'react'
import { shallow } from 'enzyme'
import { List } from './index'

describe('components/list', () => {
  it('should render an empty list with default no result', () => {
    expect(
      shallow(
        <List
          className="event-proposals"
          array={[]}
          renderRow={(proposal) => (
            <div>{proposal}</div>
          )}
        />,
      ),
    ).toMatchSnapshot()
  })
  it('should render an empty list with custom empty', () => {
    expect(
      shallow(
        <List
          className="event-proposals"
          array={[]}
          noResult="ðŸ˜¢"
          renderRow={(proposal) => (
            <div>{proposal}</div>
          )}
        />,
      ),
    ).toMatchSnapshot()
  })
  it('should render a non-empty list', () => {
    expect(
      shallow(
        <List
          className="event-proposals"
          array={['hello']}
          renderRow={(proposal) => (
            <div key={proposal}>{proposal}</div>
          )}
        />,
      ),
    ).toMatchSnapshot()
  })
})
