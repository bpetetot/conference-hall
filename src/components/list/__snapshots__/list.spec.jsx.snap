// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/list should render a non-empty list 1`] = `
<div
  className="list event-proposals"
>
  <div
    key="hello"
  >
    hello
  </div>
</div>
`;

exports[`components/list should render an empty list with custom empty 1`] = `
<div
  className="list event-proposals"
>
  <div
    className="no-result"
  >
    ðŸ˜¢
  </div>
</div>
`;

exports[`components/list should render an empty list with default no result 1`] = `
<div
  className="list event-proposals"
>
  <div
    className="no-result"
  >
    No result
  </div>
</div>
`;
